### PR TITLE
[QA-282] : Added IPVR scripts

### PR DESCRIPTION
**Pull request by abhinaysrivastavaperf**
### What?
Added IPVR scripts

#### Changes:
- IPVR scripts to generate messages for the events `AUTH_IPV_AUTHORISATION_REQUESTED`, `F2F_YOTI_START` and `IPV_F2F_CRI_VC_CONSUMED`
- There are separate scenarios for each message type to ensure that each can have separate workload patterns for each event.
- Sample payload for all three events has been provided by Kiwi team.
- Added the parameter (SQS queue name) in perf test AWS account parameter store.

---

### Why?
To conduct initial performance tests for IPVR scenario.

---
### Tasks
- [x] Add `/perfTest/identity/kiwi/stubTxMAConsumer` to the paramater store before the PR is merged

---
### Related:
- [IPVR Performance test scenario](https://govukverify.atlassian.net/wiki/spaces/DID/pages/3346661421/Performance+Test+Scenarios#IPVR---Backend-Scenario-(Message-consumption-and-processing-from-TxMA-consumer-queue))

### CODE REVIEW

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+export function generateAuthRequest (): AUTH_IPV_AUTHORISATION_REQUESTED {
+  const timestamp = new Date().toISOString()
+  const sampleAuthRequest: AUTH_IPV_AUTHORISATION_REQUESTED = {
+    event_id: uuidv4(),
+    client_id: uuidv4(),
+    clientLandingPageUrl: 'https://www.gov.uk/request-copy-criminal-record',
+    event_name: 'AUTH_IPV_AUTHORISATION_REQUESTED',
+    rp_name: 'replay',
+    timestamp: Date.now().toString().substring(0, 10),
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:18]:
```suggestion
    timestamp: Math.floor(Date.now()/1000).toString(),
```
I think technically this shouldn't presume the length of the unix timestamp, but this current solution will work till the year 2286 - so it's probably fine

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  return sampleIPVRequest
+}
+
+function randomStringGenerator (charLen: number): string {
+  let result = ''
+  const characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' // pragma: allowlist secret
+  for (let i = charLen; i > 0; i--) result += characters[Math.round(Math.random() * (characters.length - 1))]
+  return result
+}
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:89]: We actually already have this function in the repo, the `randomString()` function from `common/utils/jslib/index.js`.

https://k6.io/docs/javascript-api/jslib/utils/randomstring/

File: deploy/template.yaml [hunk 3]
```diff
@@ -378,6 +378,7 @@ Resources:
               IDENTITY_KIWI_URL: "/perfTest/identity/kiwi/url"
               IDENTITY_KIWI_STUB_URL: "/perfTest/identity/kiwi/stubUrl"
               IDENTITY_KIWI_TARGET: "/perfTest/identity/kiwi/target"
+              IDENTITY_KIWI_STUB_SQS: "/perfTest/identity/kiwi/stubTxMAConsumer"
```
Comment (Tom-Dann) [inline deploy/template.yaml:381]: - [ ] @abhinaysrivastavaperf Add `/perfTest/identity/kiwi/stubTxMAConsumer` to the paramater store before the PR is merged

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    rp_name: 'replay',
+    timestamp: Date.now().toString().substring(0, 10),
+    timestamp_formatted: timestamp,
+    user: {
+      user_id: userID,
+      govuk_signin_journey_id: signinJourneyID
+    }
+  }
+
+  return sampleF2FRequest
```
Comment (Tom-Dann) [inline deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:46]:
```suggestion
  return {
    event_id: uuidv4(),
    client_id: uuidv4(),
    clientLandingPageUrl: 'https://www.gov.uk/request-copy-criminal-record',
    event_name: 'F2F_YOTI_START',
    rp_name: 'replay',
    timestamp: Date.now().toString().substring(0, 10),
    timestamp_formatted: timestamp,
    user: {
      user_id: userID,
      govuk_signin_journey_id: signinJourneyID
    }
  }
```
Directly returning the object should still pick up the type safety

**Review by Tom-Dann — APPROVED** [review]

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts (hunk 1 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:18]: Updated

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts (hunk 4 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:46]: Updated for all three event functions

File: deploy/template.yaml (hunk 3 above)
Comment (abhinaysrivastavaperf) [reply deploy/template.yaml:381]: Yeah this has been added already. Thank you.

File: deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts (hunk 2 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/cri-kiwi/requestGenerator/ipvrReqGen.ts:89]: Removed this function and used randomString(). Thank you

File: deploy/scripts/src/common/utils/jslib/index.d.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export const uuidv4: () => string
+export const randomString: (length: number) => string
```
Comment (Tom-Dann) [inline deploy/scripts/src/common/utils/jslib/index.d.ts:2]:
```suggestion
export const randomString: (length: number, charset?: string) => string
```
Missing an optional parameter in the type definition

**Review by Tom-Dann — CHANGES_REQUESTED** [review]

File: deploy/scripts/src/common/utils/jslib/index.d.ts (hunk 5 above)
Comment (abhinaysrivastavaperf) [reply deploy/scripts/src/common/utils/jslib/index.d.ts]: Done

**Review by Tom-Dann — APPROVED** [review]

**Pull request merged**